meta-mender-raspberrypi/recipes-mender/update-firmware-state-script/files/ArtifactInstall_Leave_50

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+set -e
+
+function trap_exit() {
+  umount --lazy /mnt/inactive_part
+}
+trap trap_exit EXIT
+
+echo "$(basename "$0") was called!" >&2
+
+# Check if mmcbkl0p2 is active part
+mount | grep /dev/mmcblk0p2
+if [ $? -eq 0 ]; then
+  inactive_part="/dev/mmcblk0p3"
+else
+  inactive_part="/dev/mmcblk0p2"
```

Comment:
This will fit the vast majority, but I would still consider a small `source ${WORKDIR}/variables.sh` and corresponding:

```bash
do_compile() {
    cat > ${WORKDIR}/variables.sh <<EOF
MENDER_ROOTFS_PART_A=${MENDER_ROOTFS_PART_A}
MENDER_ROOTFS_PART_B=${MENDER_ROOTFS_PART_B}
EOF
}
```

Reply:
Makes sense, but complicates things a bit. 

As this is something that you can enable/disable it can not be bound to the `update-firmware-state-script` recipe as the script will run on the "old" image where this might not be present.

So we would need to do one of the following:

1. Add a separate recipe that always inställes a sourcable file for Raspberry Pi boards
2. Above, but make it generic that is put it in meta-mender-core, could be part of the mender-client recipe, some update modules could benefit from this?
3. Parse `/var/lib/mender/mender.conf` to extract the necessary information

Do you have a prefererad method?

Reply:
Right, I see what you mean. A separate recipe feels very heavyweight for this. Although I like method 3, unfortunately the JSON makes it a bit tricky if `jq` isn't installed. I have a 4th suggestion then:

1. Start your script with:
   ```bash
   #!/bin/sh

   @PUT_MENDER_ROOTFS_PART_A_HERE@
   @PUT_MENDER_ROOTFS_PART_B_HERE@
   ```
2. Use a sed script to replace those during the build, instead of during execution.

Reply:
> I have a 4th suggestion then: